docs/guides/implementation-guides/how-to-implement-dependency-injection.md

```diff
@@ -7,9 +7,120 @@ import MvvmArchitectureDiagram from '/img/gitbook-import/assets/image (3) (1) (2
 
 # How To Implement Dependency Injection
 
-Content in preparation.
+:::info
+For views you can make a simple [ViewLocator](../../concepts/view-locator) which will automatically resolve the view from the Locator.
+:::
 
 This guide will show you how to use Dependency Injection (DI) with _Avalonia UI_ and the MVVM pattern. 
 
 <img src={MvvmArchitectureDiagram} alt=""/>
 
+Let's assume that your MainViewModel as a dependency on MyService. The contructor for ViewModel should looks like this:
+```csharp
+private readonly MyService _myService;
+
+public MapsViewModel(MyService myService)
+{
+    _myService = myService;
+}
+```
+These are the steps you need to do in order to resolve that dependencu using DI.
+
+## Step 1: Install the NuGet package Microsoft.Extensions.DependencyInjection
+```shell
+dotnet add package Microsoft.Extensions.DependencyInjection
```

Comment:
It's worth mentioning that there are many other DI providers available but the guide will be using Microsoft's DI. It's probably also worth mentioning that Avalonia also actually ships with the Splat DI container and it's possible to use it from user code. Although it can require additional setup to do so.

Reply:
> Avalonia also actually ships with the Splat DI container

It's not. Splat is a dependency of ReactiveUI, which is an optional choice for developers.
And Splat DI was created to serve ReactiveUI extensibility purposes, and in general is pretty poor IoC implementation. It probably shouldn't be mentioned at all.

Reply:
After MS.DI a popular second choice would be lightweight DryIoC or some older but heavier options like Autofac.
Possibly some newer source-generators based solutions could be a good choice for developers, who know what they are doing. Something like Pure.DI even has an avalonia sample - https://github.com/DevTeam/Pure.DI.

Reply:
I think it would be a fair game to provide a short example with MS.DI, and reference links to more options and detailed docs on MS.DI as well.

Reply:
Huh, I guess I was mistaken. Good to know. But yes more general information about DI as well as links to more advanced usage of MS Di would also be good.

Reply:
I've added mention to the DI you listed @maxkatz6